### PR TITLE
[task-thread-pool] new port

### PR DESCRIPTION
**Pull request by alugowski**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by alugowski** [comment]
@microsoft-github-policy-service agree

File: versions/t-/task-thread-pool.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+	"versions": [
+		{
+			"git-tree": "b64ff5ad8b40dbfe129aabe53f11b81ba71e4743",
+			"version": "1.0.7",
+			"port-version": 0
+		}
+	]
+}
```
Comment (jimwang118) [inline versions/t-/task-thread-pool.json:9]: Please add a blank line at the end of the file.

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: versions/t-/task-thread-pool.json (hunk 1 above)
Comment (alugowski) [reply versions/t-/task-thread-pool.json]: Done

**Comment by alugowski** [comment]
I've resolved the comments.

File: versions/t-/task-thread-pool.json (hunk 1 above)
Comment (dg0yt) [reply versions/t-/task-thread-pool.json]: This file used tabs...
Please use `vcpkg x-add-version [--overwrite-version] <port>`. Manual editing creates non-standard formatting and unexpected changes. In addition, that command will complain about certain problems in the manifest.

Comment (dg0yt) [reply versions/t-/task-thread-pool.json]: NB: You checked "The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result." in the initial post...

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x86-windows
x64-windows
x64-windows-static
```

File: versions/t-/task-thread-pool.json (hunk 1 above)
Comment (alugowski) [reply versions/t-/task-thread-pool.json]: I did run it. Maybe the mac version doesn't reformat that particular file? It did reformat the other .json. See:
```
~/vcpkg (task-thread-pool|✔) % ./vcpkg x-add-version --all
~/vcpkg (task-thread-pool|✔) %
~/vcpkg (task-thread-pool|✔) % ./vcpkg x-add-version --overwrite-version task-thread-pool
version 1.0.7 is already in /Users/alugowski/vcpkg/versions/t-/task-thread-pool.json
version 1.0.7 is already in /Users/alugowski/vcpkg/versions/baseline.json
No files were updated for task-thread-pool
```

So I deleted the file and regenerated it with `vcpkg x-add-version --overwrite-version task-thread-pool`. PR updated.

**Pull request merged**